File: src/main/java/com/jss/osiris/modules/osiris/invoicing/service/InvoicingBlockageService.java
package com.jss.osiris.modules.osiris.invoicing.service;

import java.util.List;

import com.jss.osiris.modules.osiris.invoicing.model.InvoicingBlockage;

public interface InvoicingBlockageService {
    public List<InvoicingBlockage> getInvoicingBlockages();

    public InvoicingBlockage getInvoicingBlockage(Integer id);

    public InvoicingBlockage addOrUpdateInvoicingBlockage(InvoicingBlockage invoicingBlockage);
}

File: client/src/app/modules/invoicing/services/invoicing.blockage.service.ts
import { HttpClient, HttpParams } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { AppRestService } from 'src/app/services/appRest.service';
import { InvoicingBlockage } from '../model/InvoicingBlockage';

@Injectable({
  providedIn: 'root'
})
export class InvoicingBlockageService extends AppRestService<InvoicingBlockage> {

  constructor(http: HttpClient) {
    super(http, "invoicing");
  }

  getInvoicingBlockages() {
    return this.getList(new HttpParams(), "invoicing-blockages");
  }

  addOrUpdateInvoicingBlockage(invoicingBlockage: InvoicingBlockage) {
    return this.addOrUpdate(new HttpParams(), "invoicing-blockage", invoicingBlockage, "Enregistré", "Erreur lors de l'enregistrement");
  }

}
